backend/packages/Upgrade/src/api/services/MoocletExperimentService.ts

```diff
@@ -282,94 +285,152 @@ export class MoocletExperimentService extends ExperimentService {
+      currentExperiment: null,
+    };
     let currentMoocletExperimentRef: MoocletExperimentRef;
 
     try {
       // 1. detect the changes we might care about
       currentMoocletExperimentRef = await this.getMoocletExperimentRefByUpgradeExperimentId(incomingExperiment.id);
-      const hasEligibleInactiveStateDesignChanges = await this.detectExperimentDesignChanges(
+
+      const currentPolicyParametersResponse = await this.moocletDataService.getPolicyParameters(
+        currentMoocletExperimentRef.policyParametersId,
+        logger
+      );
+
+      const currentOutcomeVariableResponse = await this.moocletDataService.getVariable(
+        currentMoocletExperimentRef.variableId,
+        logger
+      );
+
+      const allowedInactiveStateChanges = await this.detectExperimentDesignChanges(
         incomingExperiment,
         currentMoocletExperimentRef,
-        logger
+        currentOutcomeVariableResponse.name
       );
 
-      if (incomingExperiment.state !== EXPERIMENT_STATE.INACTIVE && hasEligibleInactiveStateDesignChanges) {
+      if (incomingExperiment.state !== EXPERIMENT_STATE.INACTIVE && allowedInactiveStateChanges) {
         const error = {
-          message: '[Mooclet Edit] Ineligible changes detected for an active Mooclet experiment',
-          changes: hasEligibleInactiveStateDesignChanges,
+          message: '[Mooclet Edit] Ineligible changes detected for an active Mooclet experiment.',
+          changes: allowedInactiveStateChanges,
         };
         logger.error({ error: JSON.stringify(error) });
         throw new Error(JSON.stringify(error));
       }
 
       logger.debug({
-        message: '[Mooclet Edit] Experiment changes affect Mooclet resources, begin sync',
-        changes: hasEligibleInactiveStateDesignChanges,
+        message: '[Mooclet Edit] Experiment changes affect Mooclet resources, begin sync.',
+        changes: allowedInactiveStateChanges,
       });
 
-      // 2. if the changes are allowed, go ahead and update the upgrade experiment first so we can abort early if there are any issues
+      const currentExperiment = await this.experimentRepository.findOne({
+        where: { id: incomingExperiment.id },
+        relations: ['conditions'],
+      });
+
+      // if the changes are allowed, go ahead and update the upgrade experiment first so we can abort early if there are any issues
       const updatedExperiment = await super.update(incomingExperiment, currentUser, logger, manager);
 
       // ---------- update Mooclet Resources ------------------------------------------
-      // create a reference to current state of conditions and policyParams in case we need to rollback
-      rollbackRef = await this.createRollbackRef(currentMoocletExperimentRef, logger);
+      // create a reference of successful changes to rollback in case of error
+      // after each successful update to mooclet resource, update this ref with the resource that was changed
```

Comment:
Everything up to here in the `try` block (fetching data, updating the UpGrade experiment) would not require any rollback of the mooclet data if it fails, since no mooclet data will have been mutated yet, right? Couldn't we just bail on those without running the rollback method (which seems like it wouldn't do anything except report that a (empty) rollback was done?

Reply:
not sure what the comment refers to, possibly an earlier commit? I'm not sure how but even though I pushed updates earlier today, your very recent comment says "outdated" here on github, and the snippets it shows in the little diff snippet aren't in the current version of the branch.